### PR TITLE
Issue66 -  Improved Error Handling and Unit Tests for check_panel.py

### LINKED ISSUE #66: check_panel.py : improve logging and add unit testing
Add unit testing to check_panel.py and improve logging

### PR DESCRIPTION
**Pull request by ASendellPrice**
This PR resolves #66, introducing several improvements to the check_panel.py script, aimed at enhancing its robustness and testability:

**Error Handling Improvements:**
- Enhanced error handling across the script to better capture and log issues that may arise during API requests or data processing.
- Error logging is now fully integrated with the centralised logging configuration system developed as part of Spring3 (see #71), ensuring consistency across modules.

**Unit Tests:**
- A full suite of unit tests has been developed for the key functions in check_panel.py, ensuring correct behaviour and error handling.
- The tests cover different scenarios and edge cases, providing better coverage and confidence in the script's reliability.
- All unit tests passed
image

**Preparation for build:**
- main.py script has been developed to handle the entry point for the application.
- setup.py has been added to facilitate package installation and configuration.

**Important Notes:**
- File Relocation: The settings.py file, which contains the logging configuration, will need to be moved to the PanelPal directory for the build to function correctly.

**PEP8:**
- check_panel.py: 
image

- main.py:
image

### CODE REVIEW

**Comment by PatrickWeller** [comment]
Hi Ash,

I've tried to get this to run but I'm struggling. I know you've moved files into a further subdirectory called PanelPal. But I'm having many issues with importing different modules when running tests and stuff.
I know we can set our pythonpath, but to make sure this is an installable piece of software I think we need to get on top of how exactly we reference other modules for import. We can't all have different python path locations or it just won't work properly when someone else installs this as a piece of software.

Hopefully we can have part of an installation be a repeatable specification of the path in some way or something. But this isn't really my expertise.

**Comment by ASendellPrice** [comment]
Hi Patrick, 
Thanks for having a look at this, sorry the changes caused so many issue when importing packages. This should resolve when the PanelPal package is installed using pip. I have updated README.md to include installation instructions and included a tree showing how files will need to be structured going forwards. Let me know if you run into any other issues. 

```bash
cd PanelPal
pip install .
```

**Comment by PatrickWeller** [comment]
Hi Ash, I'll test this shortly.
Looking at the rubric we will in future want a pyproject.toml file, which i think might do all the functionality of setup.py for the installation. But I'll check to see that this all works first, and the toml file will be another issue (#73). Hopefully it doesn't require a full rework of everything and can be just adapted.

File: PanelPal/build/lib/accessories/panel_app_api_functions.py
Comment (PatrickWeller) [inline PanelPal/build/lib/accessories/panel_app_api_functions.py:1]: Add build/lib to .gitignore

**Comment by ASendellPrice** [comment]
Pull request replaced by #74 (to assist resolving branch conflicts). Closing this request.

**Pull request closed without merging**